src/element/typeChecks.ts

```diff
@@ -101,3 +103,30 @@ export const isExcalidrawElement = (element: any): boolean => {
     element?.type === "line"
   );
 };
+
+export const isRectangleElement = (
+  element: ExcalidrawElement | null,
+): element is ExcalidrawRectangleElement => {
+  return element != null && element.type === "rectangle";
+};
+
+export const hasBoundTextElement = (
+  element: ExcalidrawElement | null,
+): element is ExcalidrawBindableElement => {
+  return (
+    isBindableElement(element) &&
+    !!element.boundElements?.filter(({ type }) => type === "text").length
```

Comment:
```suggestion
    !!element.boundElements?.some((binding) => binding.type === "text")
```

Better not allocating a new array for a type check, and also a micro-perf improvement of not destructuring (even if transpiled which is an extra variable assignment).